pkg/build/strategies/strategies_test.go

```diff
@@ -0,0 +1,94 @@
+package strategies
+
+import (
+	"testing"
+
+	"github.com/openshift/source-to-image/pkg/api"
+	"github.com/openshift/source-to-image/pkg/test"
+	"github.com/openshift/source-to-image/pkg/util/user"
+)
+
+func rangeList(str string) *user.RangeList {
+	l, err := user.ParseRangeList(str)
+	if err != nil {
+		panic(err)
+	}
+	return l
+}
+
+func TestCheckAllowedUser(t *testing.T) {
+	tests := []struct {
+		name        string
+		allowedUIDs *user.RangeList
+		user        string
+		onbuild     []string
+		expectErr   bool
+	}{
+		{
+			name:        "AllowedUIDs is not set",
+			allowedUIDs: rangeList(""),
+			user:        "root",
+			onbuild:     []string{},
+			expectErr:   false,
+		},
+		{
+			name:        "AllowedUIDs is set, non-numeric user",
+			allowedUIDs: rangeList("0"),
+			user:        "default",
+			onbuild:     []string{},
+			expectErr:   true,
+		},
+		{
+			name:        "AllowedUIDs is set, user 0",
+			allowedUIDs: rangeList("1-"),
+			user:        "0",
+			onbuild:     []string{},
+			expectErr:   true,
+		},
+		{
+			name:        "AllowedUIDs is set, numeric user, non-numeric onbuild",
+			allowedUIDs: rangeList("1-10,30-"),
+			user:        "100",
+			onbuild:     []string{"COPY test test", "USER default"},
+			expectErr:   true,
+		},
+		{
+			name:        "AllowedUIDs is set, numeric user, no onbuild user directive",
+			allowedUIDs: rangeList("1-10,30-"),
+			user:        "200",
+			onbuild:     []string{"VOLUME /data"},
+			expectErr:   false,
+		},
+		{
+			name:        "AllowedUIDs is set, numeric user, onbuild numeric user directive",
+			allowedUIDs: rangeList("200,500-"),
+			user:        "200",
+			onbuild:     []string{"USER 500", "VOLUME /data"},
+			expectErr:   false,
+		},
+		{
+			name:        "AllowedUIDs is set, numeric user, onbuild user 0",
+			allowedUIDs: rangeList("1-"),
+			user:        "200",
+			onbuild:     []string{"RUN echo \"hello world\"", "USER 0"},
+			expectErr:   true,
+		},
```

Comment:
Yeah, I noticed that after scrolling down, still having those here as well doesn't hurt :wink: